code/modules/lighting/lighting_turf.dm

```diff
@@ -88,6 +88,11 @@
 			else //If fulltile and opaque, then the whole tile blocks view, no need to continue checking.
 				directional_opacity = ALL_CARDINALS
 				break
+	else
+		for(var/atom/elem in src.contents)
```

Comment:
is the assertion here that making things opaque via sources cuts off light sources, but removing it doesn't?
also consider using a signal for this instead of two for loops, it's cleaner and also a bit more performant.